File: Common/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using Common.Models.Payment;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Common.Models
{
    [Serializable]
    public record Order
    {
        [BsonId]
        [BsonGuidRepresentation(GuidRepresentation.Standard)]
        public Guid Id { get; set; } = Guid.NewGuid();

        public OrderStatus OrderStatus { get; set; } = OrderStatus.New;

        [Required]
        public required CustomerInfo CustomerInfo { get; set; }

        [Required]
        public IList<OrderItem> Items { get; set; } = new List<OrderItem>();

        [Required]
        public decimal TotalPrice { get; set; }

        [Required]
        public required Address ShippingAddress { get; set; }

        [Required]
        public required PaymentInfo PaymentInfo { get; set; }

        public decimal UseCashback { get; set; }

        public DateTime CreationDateTime { get; set; } = DateTime.UtcNow;
    }
}

File: rorderapi/src/models/order.rs
use rust_decimal::prelude::*;
use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};

use super::{address::Address, customerinfo::CustomerInfo, orderitem::OrderItem, orderstatus::OrderStatus, paymentinfo::PaymentInfo};

#[derive(Serialize, Deserialize, Clone, Debug)]
#[serde(rename_all = "PascalCase")]
pub struct Order {
    pub id: bson::Uuid,
    pub order_status: OrderStatus,
    pub customer_info: CustomerInfo,
    pub items: Vec<OrderItem>,
    pub total_price: Decimal,
    pub shipping_address: Address,
    pub payment_info: PaymentInfo,
    pub use_cashback: Decimal,
    #[serde(default = "Utc::now")]
    pub creation_date_time: DateTime<Utc>
}